### PR TITLE
Fix running Firefox wdspec tests with the correct profile

### PR DESCRIPTION
**Pull request by jgraham**
This turned into a major yak shaving exercise. Putting this PR here to run the CI and maybe ask for review later, but it's also in the Gecko code review system.

### CODE REVIEW

**Comment by jgraham** [comment]
@foolip and @gsnedders I think it makes sense to land this in mozilla-central since in pratice the failures occur in configurations we only run there, and there are associated changes in geckodriver to fix the original issue. But it's a big set of patches that shaved a lot of yaks and your input into the resulting appearance of said bovines would be much appreciated.

**Comment by foolip** [comment]
@jgraham is the best way to understand what everything is for to review each commit in isolation? Or is there a Gecko review with additional context I should look at? This is pretty big :)

**Comment by jgraham** [comment]
Yes, each commit in isolation is the right way to review; it's a few different changes to end up where we want to be. The gecko review is https://phabricator.services.mozilla.com/D113670 Note that phabricator only has a per-commit view; you need to use the stack to find the other patches (that link is to the topmost patch i.e. the last one). Very happy to get comments on Phabricator if you have an account and feel like it; the UI probably works better for this kind of change, but obviously there's no expectation of using some vendor-specific review tool :)

**Review by gsnedders — COMMENTED** [review]
Only a very quick glance over, but mostly looks sane.

File: tools/wptrunner/wptrunner/browsers/firefox.py [hunk 1]
```diff
@@ -150,7 +150,13 @@ def executor_kwargs(logger, test_type, server_config, cache_manager, run_info_da
                                   kwargs["enable_webrender"],
                                   kwargs["chaos_mode_flags"])
 
-            executor_kwargs["environ"] = environ
+        # This doesn't work with wdspec tests
+        # In particular tests can create a session without passing in the capabilites
+        # and in those cases we get the default geckodriver profile which doesn't
+        # guarantee zero network access
+        del environ["MOZ_DISABLE_NONLOCAL_CONNECTIONS"]
```
[the remark this answers is not in the document]
Comment (jgraham) [reply tools/wptrunner/wptrunner/browsers/firefox.py:193]: It's under `if test_type == "wdspec"` I think (the conditional code is getting pretty long though)

[the remark this answers is not in the document]
Comment (gsnedders) [reply tools/wptrunner/wptrunner/browsers/firefox.py:193]: Ah, yeah, it is. Sometimes it's hard to tell indentation level at a glance…

**Comment by whimboo** [comment]
@jgraham I assume that we can close this draft PR now given that the feature landed quite some time ago?

**Comment by whimboo** [comment]
> @jgraham I assume that we can close this draft PR now given that the feature landed quite some time ago?

Dupe of https://github.com/web-platform-tests/wpt/pull/29075/

**Pull request closed without merging**